dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/IndexServiceBean.java

```diff
@@ -598,21 +617,23 @@ private String addOrUpdateDataset(IndexableDataset indexableDataset) {
                 solrInputDocument.addField(SearchFields.PUBLICATION_STATUS, SearchPublicationStatus.DEACCESSIONED.getSolrValue());
             }
             Date createDate = dataset.getCreateDate();
-            if (createDate != null) {
-                String msg = "can't find major release date, using create date: " + createDate;
-                logger.fine(msg);
-                datasetSortByDate = createDate;
-            } else {
-                String msg = "can't find major release date or create date, using \"now\"";
-                logger.info(msg);
-                datasetSortByDate = new Date();
-            }
+            logger.fine("can't find major release date " +
+                    (createDate != null ? "using create date: " + createDate : "or create date, using \"now\""));
+            datasetSortByDate = createDate != null ? createDate : new Date();
         }
         solrInputDocument.addField(SearchFields.RELEASE_OR_CREATE_DATE, datasetSortByDate);
         solrInputDocument.addField(SearchFields.RELEASE_OR_CREATE_DATE_SEARCHABLE_TEXT, convertToFriendlyDate(datasetSortByDate));
 
         if (state.equals(IndexableDataset.DatasetState.PUBLISHED)) {
             solrInputDocument.addField(SearchFields.PUBLICATION_STATUS, SearchPublicationStatus.PUBLISHED.getSolrValue());
+            SolrInputField field = solrInputDocument.getField(SearchFields.PUBLICATION_STATUS);
+            if (field.getValues().stream().anyMatch(v -> SearchPublicationStatus.UNPUBLISHED.getSolrValue().equals(v))) {
+                List<Object> values = field.getValues().stream()
+                        .filter(v -> !SearchPublicationStatus.UNPUBLISHED.getSolrValue().equals(v))
+                        .collect(Collectors.toList());
+                solrInputDocument.removeField(SearchFields.PUBLICATION_STATUS);
+                solrInputDocument.addField(SearchFields.PUBLICATION_STATUS, values);
+            }
```

Comment:
Its hard for me to think up an example when this code would have any effect.
`SearchPublicationStatus.UNPUBLISHED` will be present only if indexable dataset have state `IndexableDataset.DatasetState.WORKING_COPY` and `majorVersionReleaseDate` is null.
This code runs only in indexable dataset have state `IndexableDataset.DatasetState.PUBLISHED` so it cancels each other out